pkg/server/grpc/event_data.go

```diff
@@ -700,6 +700,16 @@ func convertPacketMetadata(v *trace.PacketMetadata) (*pb.EventValue, error) {
 		}}, nil
 }
 
+// sanitizeStringArrayForProtobuf sanitizes all string elements in a slice
+// to ensure they contain only valid UTF-8 characters
+func sanitizeStringArrayForProtobuf(arr []string) []string {
+	sanitizedArray := make([]string, len(arr))
```

Comment:
Can we avoid this allocation and sanitize in place instead?

Reply:
yes, updating.